src/main/java/mate/academy/spring/model/User.java

```diff
@@ -41,10 +59,31 @@ public void setPassword(String password) {
         this.password = password;
     }
 
+    @Override
+    public boolean equals(Object o) {
+        if (this == o) {
+            return true;
+        }
+        if (!(o instanceof User)) {
+            return false;
+        }
+        User user = (User) o;
+        return Objects.equals(id, user.id) && Objects.equals(email, user.email)
+                && Objects.equals(password, user.password)
+                && Objects.equals(roles, user.roles);
+    }
+
+    @Override
+    public int hashCode() {
+        return Objects.hash(id, email, password, roles);
+    }
+
```

Comment:
It was in the checklist. Why is it a mistake? 
4. Better use Set of roles in the user model, instead of the List (don’t forget about equals and hashcode).

Reply:
Why do you think we need equals and hashcode in **User** class? That is my question) Think about it. What is the reason? 
Hint: there is some Set mentioned in the checklist.

Reply:
Got it. Thanks! Could you check my fixed request?
Thanks!